Moving the Hybrid CPUs API out of preview (#680)
oneapi-src/oneTBB: doc/main/tbb_userguide/Guiding_Task_Scheduler_Execution.rst

The processors with `Intel® Hybrid Technology <https://www.intel.com/content/www/us/en/products/docs/processors/core/core-processors-with-hybrid-technology-brief.html>`_
contain several core types, each is suited for different purposes.

> In most cases, hybrid systems don't require special API usage but in special cases, the performance

Reviewer: @aleksei-fedotov @ivankochin both are correct, but I'd rather say **in most cases** since it's more common.

> may be improved by setting the preferred core type.

Author: Just a suggestion:

In most cases, systems with hybrid CPUs shows reasonable performance without involving additional APIs calls, however in some exceptional cases, performance my by tuned by setting the preferred core type.

Reviewer 2: ```suggestion
In most cases, systems with hybrid CPU architecture show reasonable performance without involving additional API calls,
however in some exceptional situations, performance may be tuned by setting the preferred core type.
```
@ivankochin  Thank you for the suggestion :)
I guess it looks better although I made a few edits (still don't like the repeated "cases" here). What do you think?

Author: Yeah, that's better, but I would like to propose "scenarios" instead of "situations"

> To set execution preference, assign specific core type identifier to the ``task_arena::constraints::core_type`` field.

Author: Don't you think, that "execution preference" sounds too vague?

Reviewer 2: I guess it can be changed to **"To set preferred core type for the execution"** to continue the sentence above.

Reviewer 2: ```suggestion
To set preferred core type for the execution, assign specific core type identifier to the ``task_arena::constraints::core_type`` field.
```

The processors with `Intel® Hybrid Technology <https://www.intel.com/content/www/us/en/products/docs/processors/core/core-processors-with-hybrid-technology-brief.html>`_
contain several core types, each is suited for different purposes.

> In most cases, systems with hybrid CPU architecture show reasonable performance without involving additional API calls,

Reviewer: ```suggestion
In most cases, systems with hybrid CPU architecture show reasonable performance without involving additional API calls.
```

> however in some exceptional scenarios, performance may be tuned by setting the preferred core type.

Reviewer: ```suggestion
However, in some exceptional scenarios, performance may be tuned by setting the preferred core type.
```

> To set preferred core type for the execution, assign specific core type identifier to the ``task_arena::constraints::core_type`` field.

Reviewer: ```suggestion
To set the preferred core type for the execution, assign a specific core type identifier to the ``task_arena::constraints::core_type`` field.
```
